### PR TITLE
add `.choices(<options with comma-delimeter>)` helper to flag

### PR DESCRIPTION
**Pull request by maidh91**
- add `valueChoices` field the Flag class
- compare input value with value in `valueChoices`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -504,6 +510,12 @@ class Flag {
     return this
   }
 
+  choices (values) {
+    this.valueChoices = values.split(',').map(s => s.trim())
+    this.description += ` (choices: ${values})`
```
Comment (kasperisager) [inline index.js:515]:
```suggestion
    this.valueChoices = values
    this.description += ` (choices: ${values.join(', ')})`
```

No need for string parsing, `values` can just be an array of strings.

File: test.js [hunk 2]
```diff
@@ -49,6 +49,21 @@ test('command with string multiple flag', async (t) => {
   t.alike(cmd.flags.flag, ['val', 'val2'])
 })
 
+test('command with string flag with valid choice', async (t) => {
+  const cmd = command('test', flag('--flag [val] ', 'Test flag').choices('val1, val2, val3'))
```
Comment (kasperisager) [inline test.js:53]:
```suggestion
  const cmd = command('test', flag('--flag [val] ', 'Test flag').choices(['val1', 'val2', 'val3']))
```

File: test.js [hunk 3]
```diff
@@ -49,6 +49,21 @@ test('command with string multiple flag', async (t) => {
...
+  t.plan(1)
+
+  cmd.parse(['--flag', 'val2'])
+  t.is(cmd.flags.flag, 'val2')
+})
+
+test('command with string flag with invalid choice', async (t) => {
+  const cmd = command('test', flag('--flag [val] ', 'Test flag').choices('val1, val2, val3'))
```
Comment (kasperisager) [inline test.js:61]:
```suggestion
  const cmd = command('test', flag('--flag [val] ', 'Test flag').choices(['val1', 'val2', 'val3']))
```

**Review by kasperisager — CHANGES_REQUESTED** [review]

File: test.js (hunk 2 above)
Comment (maidh91) [reply test.js:53]: updated

File: test.js (hunk 3 above)
Comment (maidh91) [reply test.js:61]: updated

File: index.js (hunk 1 above)
Comment (maidh91) [reply index.js:515]: updated

**Comment by maidh91** [comment]
@kasperisager thanks for the review, I updated the code 

(I mistaken thought that `choices` is passed from CLI, but actually `choices` is defined in the code ^^)

**Review by kasperisager — APPROVED** [review]

**Pull request merged**